dashboard/app/controllers/activities_controller.rb

```diff
@@ -4,6 +4,32 @@
 
 class ActivitiesController < ApplicationController
   include LevelsHelper
+  before_filter :authenticate_user!, only: [:script_progress]
+
+  # Given a script_id return the current user's progress
+  # for each level in the script
+  #
+  # @param script_id [String] id of the script
+  # @return [JSON]
+  # {
+  #   'script_id': script_id,
+  #   'levels': {
+  #     101: 5,
+  #     ...
+  #   }
+  # }
+  def script_progress
+    script_id = params[:script_id].to_i
+    script = Script.get_from_cache(script_id)
+    raise ArgumentError, "Invalid script: #{script_id}" unless script
+
+    progress = UserProgress.script_progress(current_user, script)
```

Comment:
I think you might need to make sure this does something sensible if the user is not signed in.  (I think there are edge cases where a client page might think it was signed in but it wasn't; e.g. signing out in another tab.)